src/core/send.c

```diff
@@ -153,11 +154,20 @@ QuicSendQueueFlushForStream(
         QuicStreamAddRef(Stream, QUIC_STREAM_REF_SEND);
     }
 
-    if (Stream->Connection->State.Started) {
+    //
+    // TODO - If send was delayed by the app, under what conditions should we
+    // ignore that signal and queue anyways?
```

Comment:
I don't think we need to do this for this PR. This would be purely an optimization or performance improvement on top of this feature.

Reply:
Is this the case where a send was delayed, but then a flush still happened for whatever reason? I think it makes sense to go ahead and send the data in this case, right? [if it's as simple as that, then I don't see why this should be deferred; but if there needs to be a more subtle set of conditions, then I agree with deferral]

Reply:
I think you're asking/clarifying the following (which is true): If **anything** else causes a send to flush (on the entire connection) we're going to end up flushing the stream data too. This comment is just here to say that if we have nothing else to send and the app keeps queuing these delayed sends, it's probably a good idea to send anyways once we reach some threshold (i.e. 10k queued or something).

Reply:
Why not just send when we have a "full batch"?

Reply:
In this case, a "Full batch" would be either the max encryption batch size, or the max USO size.

Reply:
Two things:

1. There is really no such thing as a "full batch". We theoretically have a maximum batch size, but it's kind of dynamic.
2. We don't currently track "how much unsent data is currently queued", only "how much data is queued (some or all of which might be inflight already)".

Those two things make it hard to add some better logic here without further changes.

Reply:
Before we can merge, we need to make a decision on this TODO I think.